redash/query_runner/elasticsearch.py

```diff
@@ -147,7 +149,12 @@ def parse_doc(doc, path=None):
             """
             path = path or []
             result = []
-            for field, description in doc["properties"].items():
+            properties = []
+            try:
+                properties = doc["properties"].items()
+            except (TypeError, KeyError):
```

Comment:
Both `TypeError` and `KeyError` are possible in my case. The `doc` may be a string instance (and this will be a type error). Also, `doc` may be a dict without the `properties` key. I got both errors while fixing this code.